config/config.go

```diff
@@ -736,6 +736,21 @@ type MempoolConfig struct {
 	// Including space needed by encoding (one varint per transaction).
 	// XXX: Unused due to https://github.com/tendermint/tendermint/issues/5796
 	MaxBatchBytes int `mapstructure:"max_batch_bytes"`
+	// Experimental parameters to limit gossiping txs to up to the specified number of peers.
+	// This feature is only available for the default mempool (version config set to "v0").
+	// We use two independent upper values for persistent and non-persistent peers.
+	// Unconditional peers are not affected by this feature.
+	// If we are connected to more than the specified number of persistent peers, only send txs to
+	// ExperimentalMaxGossipConnectionsToPersistentPeers of them. If one of those
+	// persistent peers disconnects, activate another persistent peer.
+	// Similarly for non-persistent peers, with an upper limit of
+	// ExperimentalMaxGossipConnectionsToNonPersistentPeers.
+	// If set to 0, the feature is disabled for the corresponding group of peers, that is, the
+	// number of active connections to that group of peers is not bounded.
+	// For non-persistent peers, if enabled, a value of 10 is recommended based on experimental
+	// performance results using the default P2P configuration.
+	ExperimentalMaxGossipConnectionsToPersistentPeers    int `mapstructure:"experimental_max_gossip_connections_to_persistent_peers"`
+	ExperimentalMaxGossipConnectionsToNonPersistentPeers int `mapstructure:"experimental_max_gossip_connections_to_non_persistent_peers"`
```

Comment:
If I understand this correctly, this means that you can be connected to 20 peers but only gossip transactions (using the v0 mempool) for say 10 of those peers?

Reply:
We don't use the v0 mempool so do we still want these changes?

Reply:
> We don't use the v0 mempool so do we still want these changes?
Node operators can still use the v0 mempool. FWIW I didn't selectively exclude any commits from v0.34.35. I pulled all upstream changes (including this one). 

I don't see a strong reason to exclude this commit unless we're concerned it's dangerous or buggy. Now that you mention it, it does seem odd that an experimental feature was included in a patch release of CometBFT.